interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/create-popular-reports/create-a-hierarchical-report.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Create a Hierarchical Report
+author: Mary Sammal
+---
+# Create a Hierarchical Report
+
+This tutorial describes how to use the [detail band](..\introduction-to-banded-reports.md)'s **Hierarchy Print Options** property to create a hierarchical report.
+
+![HierarchicalReport-Result](../../../../images/eurd-HierarchicalReport-Result.png) 
+
+1. [Create a new report](../add-new-reports.md) or [open an existing one](../open-reports.md).
+
+2. [Bind the report](../bind-to-data.md) to a required data source.
+
+    The following image demonstrates an empty report bound to an [ObjectDataSource](..\bind-to-data\bind-a-report-to-an-object-data-source.md).
+
+    ![HierarchicalReport-SelectFields](../../../../images/eurd-HieararchicalReport-CreateReport.png)
+
+    Each record in this data source includes the "parent ID" field that defines the parent-child relationship and thus builds the hierarchy.
+
+3. Arrange controls on the report.
+
+    - Add the [Report Header](..\introduction-to-banded-reports.md) and [Page Header](..\introduction-to-banded-reports.md) bands (see the **Manage Report Bands | Add Bands** section in the [Introduction to Banded Reports](..\introduction-to-banded-reports.md) document for details)
+    - Add [data-bound labels](..\use-report-elements\use-basic-report-controls\label.md) to the **Detail** band.
+
+    ![HierarchicalReport-ArrangeControls](../../../../images/eurd-win-HierarchicalReport-ArrangeControls.png)
+
+    Switch to the [Preview](..\preview-print-and-export-reports.md) tab to see an intermediate result.
+
+    ![HierarchicalReport-ArrangeControls-Result](../../../../images/eurd-win-HierarchicalReport-ArrangeControls-Result.png)
+
+4. Specify the Detail band's **Hierarchy Print Options** property.
+
+    ![HierarchicalReport-Set-HierarchyPrintOptions](../../../../images/eurd-win-HierarchicalReport-Set-HierarchyPrintOptions.png)
+
+    Set the following options:
+
+    - **Key Field Name** and **Parent Field Name**, or **Child List Field Name**  
+    Set the **Key Field Name** and **Parent Field Name** properties if your report's data have the Id-ParentID related fields.  
+    Set the **Child List Field Name** property if your report's data is recursive. Assign the collection of child objects (records) if they have the same type as the parent objects (records).
+    
+    - **Indent**   
+    Specify the child level node offset.
+
+    - **Keep Together with First Child**  
+    Specify whether to print a parent node together with its first child node on the next page, if these nodes do not fit at the end of a page.
```

Comment:
Specify whether or not to print a parent node together with its first child node on the next page if these nodes do not fit at the end of a page.